addons/danger/functions/fnc_tacticsSuppress.sqf

```diff
@@ -0,0 +1,96 @@
+#include "script_component.hpp"
+/*
+ * Author: nkenny
+ * Leader calls for extended suppressive fire towards buildings or location
+ *
+ * Arguments:
+ * 0: Group leader <OBJECT>
+ * 1: Group threat unit <OBJECT> or position <ARRAY>
+ * 2: Units in group, default all <ARRAY>
+ *
+ * Return Value:
+ * success
+ *
+ * Example:
+ * [bob, angryJoe] call lambs_danger_fnc_tacticsSuppress;
+ *
+ * Public: No
+*/
+params ["_unit", "_target", ["_units", []], ["_delay", 12]];
+
+// find target
+_target = _target call CBA_fnc_getPos;
+
+// reset tactics
+[
+    {
+        params ["_group", "_enableAttack"];
+        if (!isNull _group) then {
+            _group setVariable [QGVAR(tactics), nil];
+            _group enableAttack _enableAttack;
+            _group setVariable [QGVAR(tacticsTask), nil];
+        };
+    },
+    [group _unit, attackEnabled _unit],
+    _delay
+] call CBA_fnc_waitAndExecute;
+
+// alive unit
+if !(_unit call EFUNC(main,isAlive)) exitWith {false};
+
+// clear attacks!
+{
+    if (currentCommand _x isEqualTo "ATTACK") then {
+        _x forgetTarget (assignedTarget _x);
+    };
+    _x setUnitPosWeak "MIDDLE";
+} foreach units _unit;
+
+// find units
+if (_units isEqualTo []) then {
+    _units = [_unit] call EFUNC(main,findReadyUnits);
+};
+if (_units isEqualTo []) exitWith {false};
+
+// find vehicles
+private _vehicles = [];
+{
+    if (!(isNull objectParent _x) && { isTouchingGround vehicle _x } && { canFire vehicle _x }) then {
+        _vehicles pushBackUnique vehicle _x;
+    };
+} foreach (units _unit select { _unit distance2D _x < 350 && { canFire _x }});
```

Comment:
the select can be put into the if statement we don't need to iterate 2x the same array

I don't know if a select, apply, and append + arrayIntersect be more readable / faster? @diwako?